doc/performance-investigations/0018-pdf-weasyprint-pdfkit/summary.md

```diff
@@ -0,0 +1,113 @@
+  The POC delivered is mostly 1-2 seconds. WKHTML seems to only be slow on the initial download request as it might be doing some caching or the library binaries don’t need reloading, so subsequent download appears to be faster. This may even apply between instances/users. In my testing, WKHTML was only sometimes slow. In most cases, it was on par with weasyprint.
+
+This documentation benchmarks Weasyprint and PDFKit under higher loads to see the performance differences and whether the libraries take advantage of some caching.
+
+## Runner settings
+
+| Setting       | Value                                    |  
+|---------------|------------------------------------------|  
+| Concurrency   | 115                                      |  
+| Max instances | 3                                        |  
+| Min instances | 3                                        |  
+| CPU           | 4                                        |  
+| Memory        | 4G                                       |  
+| Commit        | 4e810eb782d0614563a13844def69dd42b1770b3 |  
+
+When testing PDFKit with a custom CSS file, commit `03f3bccbd37e4f31d9621137dc8ba12ce5dc3b0b`  was used, which implemented this. This should have no impact on the other test runs.
+
+## Benchmark profile
+
+| Option                 | Value                                                                                                   |  
+|------------------------|---------------------------------------------------------------------------------------------------------|  
+| Requests file          | test_business_benchmark.json / test_business_benchmark_weasy.json / test_business_benchmark_pdfkit.json |  
+| Run time               | 10m                                                                                                     |  
+| User wait time minimum | 1                                                                                                       |  
+| User wait time maximum | 2                                                                                                       |  
+| Clients                | 50                                                                                                      |  
+| Hatch rate             | 50                                                                                                      |  
+
+## Results
+
+| Environment                                              | Requests per second | GETs (99th) (ms) | POSTs (99th) (ms) | PDF GETs (50th) (ms) | PDF GETs (90th) (ms) | PDF GETs (95th) (ms) | PDF GETs (99th) (ms) | PDF GETs (99.9th) (ms) | PDF GETs (100th) (ms) | 99th percentile Max CPU Usage (%) | Total Requests | Total Failures | Outputs                                              |  
```

Comment:
There is a lot of information here, and hard to analyse. I think we need to slim it down, the other performance tests we just have the 99th percentile. Not against a 50th and 99th, but a 90th, 95th, 99th, 99.9th can be removed in my opinion. You have included those results elsewhere if someone is really keen

Reply:
I think it was good to see both ends of the scale for the PDF. I'm happy either way. Will see what the other reviews think also, and then remove if needed.

Reply:
I'll slim down by getting rid of 50th, 90th, 99.9th. I think having 95th, 99th and maybe 100th is useful since we are looking at a specific endpoint.

Reply:
Okay, in that case I'll just keep it given it's 2 for and 2 against. I thought it was useful, but I do see why it can look daunting.